cloudant-client/src/test/java/com/cloudant/tests/ReplicationMockTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright © 2018 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.tests;
+
+import static com.cloudant.tests.HttpTest.takeN;
+import static com.cloudant.tests.util.MockWebServerResources.EXPECTED_OK_COOKIE;
+import static com.cloudant.tests.util.MockWebServerResources.EXPECTED_OK_COOKIE_2;
+import static com.cloudant.tests.util.MockWebServerResources.JSON_OK;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+
+import com.cloudant.client.api.CloudantClient;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import okhttp3.mockwebserver.MockWebServer;
+import okhttp3.mockwebserver.RecordedRequest;
+
+import java.nio.charset.Charset;
+
+/**
+ * Created by samsmith on 09/03/2018.
+ */
+
+public class ReplicationMockTest {
+
+    @Rule
+    public MockWebServer mockWebServer = new MockWebServer();
+
+    final private static String authJson = "\"auth\":{\"iam\":{\"api_key\":\"";
+    final private static String sourceDbUrl = "https://foo.cloudant.com/source";
+    final private static String targetDbUrl = "https://bar.cloudant.com/target";
+
+    @Test
+    public void createAndTriggerReplicationWithNoIamAuth() throws Exception {
+        mockWebServer.enqueue(JSON_OK);
+
+        CloudantClient c = CloudantClientHelper.newMockWebServerClientBuilder(mockWebServer)
+                .build();
+
+        c.replication()
+                .source(sourceDbUrl)
+                .target(targetDbUrl)
+                .trigger();
+
+        RecordedRequest[] requests = takeN(mockWebServer, 1);
+
+        assertEquals(requests[0].getPath(), "/_replicate");
+
+        String body = requests[0].getBody().readString(Charset.forName("UTF-8"));
```

Comment:
this can be simplified throughout to:
`getBody().readUtf8()`

Reply:
@smithsz did this get lost somewhere?

Reply:
Whoops. Must have missed these.